nova/virt/vmwareapi/vm_util.py

```diff
@@ -1434,47 +1436,47 @@ def get_vm_state(session, instance):
     return constants.POWER_STATES[vm_state]
 
 
-def _get_host_reservations(host_reservations_map, host_moref, host_vcpus,
-                           host_memory_mb):
+def _set_host_reservations(stats, host_reservations_map, host_moref):
     """Compute the number of vcpus and memory in MB reserved for the host from
     the configured reservations or the default.
 
     For every host, both vcpus and memory can be given as static number or in
     percent, with static number taking priority.
     """
-    reservations = {
-        'vcpus': 0,
-        'memory_mb': 0
-    }
+    host_vcpus = stats["vcpus"]
+    host_memory_mb = stats["memory_mb"]
+    stats["vcpus_reserved"] = 0
+    stats["memory_mb_reserved"] = 0
```

Comment:
I don't really like how we hide those parameters now, that docstring talks about. I would have probably opted for a `stats.update(_get_host_reservations(…))` instead and changed the returned dict to match the expected keys.

Reply:
It is rather a microoptimisation, I admit. I do not like to create a temporary dictionary to hold just two values, just to merge it right afterwards in a single callers dict.